lib/guess_html_encoding.rb

```diff
@@ -45,7 +45,7 @@ def self.encode(html, headers = nil)
     if html_copy.valid_encoding?
       html_copy
     else
-      html_copy.force_encoding('ASCII-8BIT').encode('UTF-8', :undef => :replace, :invalid => :replace)
```

Comment:
The idea was that if the encoding is invalid, we've probably mis-guessed, so we assume ascii. Your new code assumes that our guess is right, and that there was a mistake in the encoding.  I'm not sure which is better.